torch/csrc/utils/python_arg_parser.h

```diff
@@ -250,17 +251,20 @@ inline c10::optional<at::Scalar> PythonArgs::scalarOptional(int i) {
 inline std::vector<at::Tensor> PythonArgs::tensorlist(int i) {
   if (!args[i]) return std::vector<at::Tensor>();
   PyObject* arg = args[i];
-  auto tuple = PyTuple_Check(arg);
+  auto tuple = six::isTuple(arg);
+  if (tuple) arg = six::toTuple(arg);
```

Comment:
Could we make this `THPObjectPtr six::asTuple`? That way you get reference counting for free, and you can just wrap everything inside an if.

Reply:
+1 
You will need to wrap arg as well, but this will save you from all `if (tuple) Py_DECREF(arg);`